### ISSUE TITLE
Add AutoAugmentation to image classification training

### ISSUE THREAD

**Issue by saad-palapa** [ludwig-ai/ludwig#3581]
**Is your feature request related to a problem? Please describe.**

The image augmentation capabilities currently offered in this repository have room for enhancement, particularly given the various advanced techniques available in recent research.

Recent research papers have demonstrated superior techniques that could be beneficial:
[AutoAugment: Learning Augmentation Policies from Data](https://arxiv.org/abs/1805.09501)
[RandAugment: Practical automated data augmentation with a reduced search space](https://arxiv.org/abs/1909.13719)
[TrivialAugment: Tuning-free Yet State-of-the-Art Data Augmentation](https://arxiv.org/abs/2103.10158)
[AugMix: A Simple Data Processing Method to Improve Robustness and Uncertainty](https://arxiv.org/abs/1912.02781)

**Describe the use case**

Image augmentation aids in creating a richer training dataset via image transformations. While the current features serve the purpose, the integration of state-of-the-art AutoAugmentation techniques would elevate the capabilities of Ludwig.

**Describe the solution you'd like**

I recommend looking into the following libraries and solutions to integrate:
- [Torchvision](https://pytorch.org/vision/stable/transforms.html#auto-augmentation)
- [Google Brain AutoML](https://github.com/google/automl/blob/master/efficientdet/aug/autoaugment.py)

**Describe alternatives you've considered**

Checkout this list of [data augmentation libraries](https://github.com/AgaMiko/data-augmentation-review#computer-vision)

**Comment by tgaddair** [comment]
Thanks for raising this, @saad-palapa. Is there a particular technique you'd be most interested in having added to Ludwig? And is this something you'd be interested in helping to contribute?

**Comment by saad-palapa** [comment]
[TrivialAugment](https://pytorch.org/vision/stable/generated/torchvision.transforms.v2.TrivialAugmentWide.html#torchvision.transforms.v2.TrivialAugmentWide) seems like a good fit because of its simplicity and performance.

Take a look at Nvidia's blog post:
[Why Automatic Augmentation Matters](https://developer.nvidia.com/blog/why-automatic-augmentation-matters/)

This stuff is really exciting to me but I don't have capacity to submit PRs at the moment. However, I would love to be part of the conversation.

**Comment by tgaddair** [comment]
Thanks @saad-palapa, happy to take a look at the complexity to integrate this, hopefully it should be pretty straightforward and we can do it when some spare cycles open up.

**Comment by saad-palapa** [comment]
@skanjila, let's work on TrivialAugment for our first implementation

**Comment by jimthompson5802** [comment]
@saad-palapa @skanjila  I took a look at the TrivialAugmentation api.  A few things I noticed:
* it appear to have been introduced in torchvision 0.15
* There is a v1 and v2 version of the API, with v2 marked as Beta
* The latest version of torchvision is 0.16...v2.TrivialAugmentation is still marked as Beta.
* Current CI test is based on torchvison 0.15

I'll take stab a pulling together a PR for Trivial Augmentation.

We'll need to assess the impact of torchvision 0.16 and the v1 vs v2 api and the Beta status.  The torchvision project is recommending using v2.
